ambry-cloud/src/main/java/com/github/ambry/cloud/azure/AzureCloudDestinationSync.java

```diff
@@ -295,10 +300,88 @@ public CompletableFuture<Void> downloadBlobAsync(BlobId blobId, OutputStream out
+    BlobClient blobClient = createOrGetBlobStore(blobLayout.containerName).getBlobClient(blobLayout.blobFilePath);
+    BlobRequestConditions blobRequestConditions = new BlobRequestConditions().setIfMatch(blobProperties.getETag());
+    return blobClient.setMetadataWithResponse(metadata, blobRequestConditions,
+        Duration.ofMillis(cloudConfig.cloudRequestTimeout), Context.NONE);
+  }
+
+  /**
+   * Returns blob properties from Azure
+   * @param blobLayout BlobLayout
+   * @return Blob properties
+   * @throws CloudStorageException
+   */
+  protected BlobProperties getBlobProperties(AzureBlobLayoutStrategy.BlobLayout blobLayout)
+      throws CloudStorageException {
+    try {
+      return createOrGetBlobStore(blobLayout.containerName).getBlobClient(blobLayout.blobFilePath).getProperties();
+    } catch (Throwable t) {
+      // Due to legacy CloudBlobStore, we have to handle 404 as a special case when it can actually be a general case.
+      if (t instanceof BlobStorageException
+          && ((BlobStorageException) t).getErrorCode() == BlobErrorCode.BLOB_NOT_FOUND) {
+        String error = String.format("Failed to get blob properties for %s from Azure blob storage because it does not exist", blobLayout);
+        logger.error(error);
+        throw new CloudStorageException(error, new StoreException(error, StoreErrorCodes.ID_Not_Found), CloudBlobStore.STATUS_NOT_FOUND, false, null);
+      }
+      String error = String.format("Failed to get blob properties for %s from Azure blob storage because %s", blobLayout, t.getMessage());
+      logger.error(error);
+      throw new CloudStorageException(error, new StoreException(error, StoreErrorCodes.Unknown_Error));
+    }
+  }
+
   @Override
   public boolean deleteBlob(BlobId blobId, long deletionTime, short lifeVersion,
-      CloudUpdateValidator cloudUpdateValidator) throws CloudStorageException {
-    return false;
+      CloudUpdateValidator unused) throws CloudStorageException {
+    Timer.Context storageTimer = azureMetrics.blobUpdateDeleteTimeLatency.time();
+    AzureBlobLayoutStrategy.BlobLayout blobLayout = azureBlobLayoutStrategy.getDataBlobLayout(blobId);
+    String blobIdStr = blobLayout.blobFilePath;
+    Map<String, String> newMetadata = new HashMap<>();
+    newMetadata.put(CloudBlobMetadata.FIELD_DELETION_TIME, String.valueOf(deletionTime));
+    newMetadata.put(CloudBlobMetadata.FIELD_LIFE_VERSION, String.valueOf(lifeVersion));
+    BlobProperties blobProperties = getBlobProperties(blobLayout);
+    Map<String, String> cloudMetadata = blobProperties.getMetadata();
+    // lifeVersion must always be present
+    short cloudlifeVersion = Short.parseShort(cloudMetadata.get(CloudBlobMetadata.FIELD_LIFE_VERSION));
+    // This is the correct behavior. For ref, look at BlobStore::delete and ReplicaThread::applyUpdatesToBlobInLocalStore
+    if (cloudlifeVersion == lifeVersion && cloudMetadata.containsKey(CloudBlobMetadata.FIELD_DELETION_TIME)
```

Comment:
IMHO, I think you could reuse V1 `CloudUpdateValidator.validateUpdate` and its existing implementation `preDeleteValidation` to check if this update is valid. For example, `preDeleteValidation` also checks if account and container stored in azure metadata matches with blob metadata via `validateAccountAndContainer`.

Reply:
done